### PR TITLE
API hints

### PR DESCRIPTION
**Pull request by kecnry**
Implement API hints across all plugins.

~Requires https://github.com/spacetelescope/jdaviz/pull/3159 to be merged and included in the 4.0 release of jdaviz.~

**Waiting for**: ~jdaviz 4.0~

### CODE REVIEW

File: lcviz/plugins/ephemeris/ephemeris.vue [hunk 1]
```diff
@@ -201,8 +237,13 @@
...
             :spinner="query_spinner"
             :results_isolated_to_plugin="false"
+            :api_hints_enabled="api_hints_enabled"
             @click="query_for_ephemeris">
-              Query
+              {{ api_hints_enabled ? 
+                'plg.query_for_ephemeris()'
+                :
+                'Query'
+              }}
```
Comment (bmorris3) [inline lcviz/plugins/ephemeris/ephemeris.vue:246]: You got the `query_for_ephemeris` method for this portion of the plugin, but not the name/RA/Dec/radius fields above:

Screenshot 2024-08-21 at 14 44 28

**Review by bmorris3 — APPROVED** [review]
This looks great. I'm approving, with the caveat of the one comment below.

File: lcviz/plugins/ephemeris/ephemeris.vue (hunk 1 above)
Comment (kecnry) [reply lcviz/plugins/ephemeris/ephemeris.vue:246]: yep, you didn't expose those to the API in #127 😝   Do you want me to add them here?

Comment (kecnry) [reply lcviz/plugins/ephemeris/ephemeris.vue:250]: these are now exposed in the API and hints added in this PR.

**Pull request merged**